mailpoet/lib/Segments/DynamicSegments/Filters/MailPoetCustomFields.php

```diff
@@ -88,30 +88,30 @@ private function applyForDate(QueryBuilder $queryBuilder, DynamicSegmentFilterEn
   }
 
   private function applyForDateMonth(QueryBuilder $queryBuilder, string $valueParam): QueryBuilder {
-    $queryBuilder->andWhere("month(subscribers_custom_field.value) = month($valueParam)");
+    $queryBuilder->andWhere("month(subscribers_custom_field.value) = month(:$valueParam)");
```

Comment:
I see that in other places you're removing colons, but here and below you're adding them instead, I'd like to get some context about this.

Reply:
I had to change this in more places, and it depends on the context. Because we used a newer version of Doctrine DBAL, a parameter name with the colon in a name in the method `setParameter()` is no longer allowed. But when we use named parameters in SQL, we still need to use the colon here because it identifies the parameter.

For example, when we have a parameter name stored in a variable `$parameter = ':param'` in the past it was no issue when we used it in SQL `WHERE s.name = $parameter` and then used `$qb->setParameter($parameter)`. With the version DBAL 3 we cannot do this anymore, and I have to remove the colon from variables and add it before them in SQL. So, now it must be: `$parameter = 'param'`, `WHERE s.name = :$parameter`.

I hope this will help you to understand those changes better. 

I'm sorry I should add it to the PR description.

Reply:
Thanks for the explanation. This is a bit of a gotcha, I wonder if it could be documented somewhere so there would be less risk of a parameter with an invalid name quietly slipping into code when other developers work on the codebase. What happens if you forget to put a colon in SQL before a variable?

Reply:
> This is a bit of a gotcha, I wonder if it could be documented somewhere so there would be less risk of a parameter with an invalid name quietly slipping into code when other developers work on the codebase. What happens if you forget to put a colon in SQL before a variable?

It took me some time to find out why some of our queries stopped working. I tried to find a page about it in the documentation, but I was not successful.
When you forget to put the colon in a query, it will suspect that it is not a variable but the column's name. When you use the colon in a parameter name, it will throw an exception about a missing value for the parameter.

TLDR: Your query will not work.

It's interesting that a part of our code was without an issue, but somewhere, we needed to fix it. I think we don't need to mention anything special in our documentation.

Reply:
> When you forget to put the colon in a query, it will suspect that it is not a variable but the column's name

I assume it will throw an error in this case. Is it clear enough so it can be quickly fixed? Or will it require lengthy debugging?

Reply:
> I assume it will throw an error in this case. Is it clear enough so it can be quickly fixed? Or will it require lengthy debugging?

The exception contains `Named parameter "email_0" does not have a bound value.`. I think that's a pretty good guide. 
What do you think?